### DISCUSSION TITLE
Some questions about saturation states

### DISCUSSION THREAD

**Discussion by Tianmeng818** [PMEAL/OpenPNM#2760] [Q&A]
Dear Doctor:
Hello! Previously in studying this article of yours from 2016 "OpenPNM: A Pore Network Modeling Package", I have successfully reproduced it in versions 2.8.2 and 3.1.2, at least without reporting errors. However, there have been a few questions that have been bothering me and I would like to get your help.The following questions are based on version 2.8.2:

1.In this 2016 article, the final effect image is not sure how you implemented it in paraview, because the effect image I generated does not match your original image, I can only generate simple water intrusion or gas escape, not combine them, while your original image combines them.
![6DFMTAJ8E_1FP31SOGWW(PH](https://github.com/PMEAL/OpenPNM/assets/135722050/7695319c-15bb-4bf0-9ace-5da784095c28)

2.In this article in 2016, about the ">8000" in this line of code, I don't quite understand it, after deleting these two lines of code, the program can run, so I don't know its exact meaning, I hope to get your answer.
phys_air [ 'throat.conductance'] = phys_air [ 'throat.diffusive_conductance '] '(OP [ 'throat.inv_Pc ] 〉 8000)

My guess is that it has something to do with the saturation state, and that this code can be applied to implement a model with a non-saturated initial state, but I have been trying to find that the function I want is not possible. My attempt is to change the value ">8000" to extract a different conductivity, and changing 8000 to 12000 will turn part of the phys_air['throat.conductance'] value to 0, thus reporting an error.
![6PJMI~$8W9ENVFQUGEWBO56](https://github.com/PMEAL/OpenPNM/assets/135722050/afd1936d-bb44-4409-b193-2fb2f1c08217)
Deleting the zero element in phys_air['throat.conductance'], but cannot assign the array a to phys_air['throat.conductance'], will report an error.
![E28REZF3SW3SBKLUS5$JMK8](https://github.com/PMEAL/OpenPNM/assets/135722050/d64399dc-c2b5-4627-accc-c28775497887)
So I'm not sure if my guess "the use of this line of code is related to the saturation state of the implementation model" is correct, and would appreciate your answer.Maybe you have another method or example to create a non-saturated initial state?

3.So I tried to use other methods to implement the initial state of non-saturation, based on your 2016 code, I added an attempt to set the initial saturation state by fixing the initial saturation and marking pore.occupancy as 1, indicating that the pore is occupied.
![W O@Y9XW 6YJRS9DH`}UYAK](https://github.com/PMEAL/OpenPNM/assets/135722050/01c4bde2-c883-47bb-80bf-6377c1514b0c)
An error is reported after running the code.
![{HL _ )PJ$VQ })7$MUOD9J](https://github.com/PMEAL/OpenPNM/assets/135722050/95ba0d10-55a1-43cd-b64c-5af78b95ea6a)
So I would like to ask what is the solution for this error report? And is there any example to build a non-saturated initial state to follow?

4.Now I try to implement in openpnm2.8.2 version, first the water body invades from the upper part to the lower part, then the gas invades from the upper part to the lower part, and then after the whole process, finally get the state at a certain moment in the whole process, similar to doing different time slices for the model, I want to try the time step related method, but there is no example so I haven't implemented it yet. I don't know if there are any relevant examples to learn.

Sorry if I am disturbing you, but this has really bothered me for a long time and I hope you can help me sort it out.

**Comment by jgostick** [comment]
The software continues to evolve with time, so I am not surprized that old codes does not work with the new versions.  We try to collect useful examples on the website...have you searched through all of those already?  If you cannot find one that explains what you're doing, then let me know and I'll add it to my to-do list to write one.

**Reply by Tianmeng818** [reply]
Thanks for your reply, I hope I'm not disturbing you. I'm trying to learn openpnm from your earliest article, I'm interested in the water-driven gas and gas-driven water simulations in your 2016 article, and since I don't have any more code or examples to learn from, I'd like to try to reproduce it first and get what I can learn from it. I have tried for a long time to solve the ideas and problems I mentioned above, and tried to find examples in the website to learn from, but I didn't find anything, so I am looking for your help. Thank you for your help！

**Reply by jgostick** [reply]
Forget trying to reproduce that article, it didn't have any meaningful data in it anyway.  Here is a modern example of what you're looking for: https://openpnm.org/examples/applications/relative_diffusivity.html

**Reply by Tianmeng818** [reply]
Dear Professor, I am sorry to bother you again, I am still working on the problem about saturation recently. During my study, I found that when I use InvasonPercolation or OrdinaryPercolation algorithm for water intrusion, the default initial state is dry. If I need to implement water intrusion for a pore network model with an initial state of 50% saturation, I wonder if there is an example for reference? After a few weeks of thinking about it, 1. it is not clear how to implement the extraction of the initial state of 50% saturation, and 2. it is not clear how to implement the water intrusion for a medium with an initial state of 50% saturation. I hope you can help answer my question.
